python/ray/air/config.py

```diff
@@ -146,34 +145,44 @@ class DatasetConfig:
     split: Optional[bool] = None
 
     # Whether to raise an error if the Dataset isn't provided by the user.
-    # True by default for the "train" dataset only.
+    # False by default.
     required: Optional[bool] = None
 
-    # Whether the dataset can be streamed into memory using pipelined reads.
-    # When enabled, get_dataset_shard() returns DatasetPipeline instead of Dataset.
-    # Note that streaming isn't enabled unless you set stream_window_size too.
-    # False by default for all datasets.
-    streamable: Optional[bool] = None
-
     # Whether to transform the dataset with the fitted preprocessor. This must be
     # enabled at least for the dataset that is fit.
-    # True by default for all datasets.
+    # True by default.
     transform: Optional[bool] = None
 
-    # List of fields that the user cannot override in ``dataset_config``.
-    _noncustomizable_fields: Tuple[str] = ()
-
-    # TODO: add stream_window_size, global/local shuffle options
+    # Whether the dataset should be streamed into memory using pipelined reads.
+    # When enabled, get_dataset_shard() returns DatasetPipeline instead of Dataset.
+    # The amount of memory to use is controlled by `stream_window_size`.
+    # False by default.
+    use_stream_api: Optional[bool] = None
+
+    # Configure the streaming window size in bytes. A good value is something like
+    # 20% of object store memory. If set to -1, then an infinite window size will be
+    # used (similar to bulk ingest). This only has an effect if use_stream_api is set.
+    # Set to 1.0 GiB by default.
+    stream_window_size: Optional[float] = None
+
+    # Whether to enable global shuffle (per pipeline window in streaming mode). Note
+    # that this is an expensive all-to-all operation, and most likely you want to use
+    # local shuffle instead.
```

Comment:
We should create a shuffling feature guide that we can link to, one that expands on this FAQ: https://docs.ray.io/en/master/data/faq.html#when-should-i-use-global-per-epoch-shuffling

Reply:
Added link.